client-core/src/service/wallet_service.rs

```diff
@@ -109,7 +114,11 @@ pub struct WalletInfo {
     /// passphrase used when import wallet
     pub passphrase: Option<SecUtf8>,
     ///public_key -> encoded private_key pairs, private key is None for hardware wallet
-    pub key_pairs: BTreeMap<PublicKey, Option<Vec<u8>>>,
+    pub key_pairs: BTreeMap<PublicKey, Vec<u8>>,
+    /// public_key -> hd path pairs for hardware wallet
+    pub key_chainpath: BTreeMap<PublicKey, String>,
+    /// hdkey for hd wallet and hw wallet
+    pub hdkey: Option<Vec<u8>>,
```

Comment:
I had to lookup what this is supposed to be -- so I assume it's that "struct" HdKey that has latest "change" address index? Why is it as `Vec<u8>` and not directly `HdKey`?
Also one thing to add a comment / explain regarding HW wallet -- there's a "seed" which won't be relevant to HW wallet (as the seed is on the device) -- so I assume here it's a seed that's different from HW wallet, only used for view keys / the mixed mode?

Reply:
I use`Vec<u8>` here because if I use sturct `HdKey` when export wallet,  I first get it from storage which is `Vec<u8>`, then decode to `HdKey`,  then encode to `Vec<u8>` to print out or store to a file, when import wallet, I should do decode/encode  to. If I use `Vec<u8>` directly, it's not need to do encode/decode.